File: service/jvm-quarkus/src/main/java/com/ecobenchmark/controllers/getlists/ListResponse.java
package com.ecobenchmark.controllers.getlists;



import java.time.Instant;
import java.util.List;
import java.util.UUID;

public class ListResponse {

    private UUID id;
    private String name;
    private List<TaskResponse> tasks;
    private Instant creationDate;
    private UUID accountId;

    public ListResponse(UUID id, String name, Instant creationDate, UUID accountId) {
        this.id = id;
        this.name = name;
        this.creationDate = creationDate;
        this.accountId = accountId;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<TaskResponse> getTasks() {
        return tasks;
    }

    public void setTasks(List<TaskResponse> tasks) {
        this.tasks = tasks;
    }

    public Instant getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Instant creationDate) {
        this.creationDate = creationDate;
    }

    public UUID getAccountId() {
        return accountId;
    }

    public void setAccountId(UUID accountId) {
        this.accountId = accountId;
    }
}

File: service/php-symfony-nginx/src/Controller/GetLists/ListResponse.php
<?php

namespace App\Controller\GetLists;

use App\Entity\Account;
use App\Entity\ListEntity;

class ListResponse
{
    public string $id;
    public string $name;
    public array $tasks;
    public string $creation_date;
    public string $account_id;

    /**
     * @param ListEntity[] $listEntities
     * @return array
     */
    static function fromListEntities(array $listEntities): array
    {
        $listResponses = [];
        foreach ($listEntities as $list) {
            $listResponses[] = self::fromListEntity($list);
        }
        return $listResponses;
    }

    static function fromListEntity(ListEntity $list): ListResponse
    {
        $listResponse = new ListResponse();
        $listResponse->id = $list->getId();
        $listResponse->name = $list->getName();
        $listResponse->creation_date = $list->getCreationDate()->format('c');
        $listResponse->account_id = $list->getAccount()->getId();

        foreach ($list->getTasks() as $task) {
            $listResponse->tasks[] = TaskResponse::fromTaskEntity($task);
        }

        return $listResponse;
    }
}
